Xamarin.Forms.Core/ToolbarItem.cs

```diff
@@ -12,11 +12,34 @@ public class ToolbarItem : MenuItem
 
 		static readonly BindableProperty PriorityProperty = BindableProperty.Create("Priority", typeof(int), typeof(ToolbarItem), 0);
 
+		public static readonly BindableProperty IsVisibleProperty = BindableProperty.Create("IsVisible", typeof(bool), typeof(ToolbarItem), true, BindingMode.TwoWay, propertyChanged: (bindable, oldvalue, newvalue) =>
+		{
+			var item = bindable as ToolbarItem;
+
+			if (item != null)
+			{
+				if (item.Parent == null)
+					return;
+
+				var items = ((Page)item.Parent).ToolbarItems;
+
+				if ((bool)newvalue && !items.Contains(item))
+				{
+					Device.BeginInvokeOnMainThread(() => { items.Add(item); });
+				}
+				else if (!(bool)newvalue && items.Contains(item))
+				{
+					Device.BeginInvokeOnMainThread(() => { items.Remove(item); });
+				}
+			}
+		});
+
 		public ToolbarItem()
 		{
+	
 		}
 
-		public ToolbarItem(string name, string icon, Action activated, ToolbarItemOrder order = ToolbarItemOrder.Default, int priority = 0)
+		public ToolbarItem(string name, string icon, Action activated, ToolbarItemOrder order = ToolbarItemOrder.Default, int priority = 0, bool isVisible = true)
```

Comment:
this is an ABI change and will break on already compiled code. please add an overload with the old API